core/src/main/java/org/geysermc/geyser/translator/protocol/java/JavaUpdateRecipesTranslator.java

```diff
@@ -214,23 +229,27 @@ public void translate(GeyserSession session, ClientboundUpdateRecipesPacket pack
             }
         }
 
-        // FIXME: if the server/viaversion doesn't send trim recipes then we shouldn't either.
-
-        // BDS sends armor trim templates and materials before the CraftingDataPacket
-        TrimDataPacket trimDataPacket = new TrimDataPacket();
-        trimDataPacket.getPatterns().addAll(TrimRecipe.PATTERNS);
-        trimDataPacket.getMaterials().addAll(TrimRecipe.MATERIALS);
-        session.sendUpstreamPacket(trimDataPacket);
+        session.getLastRecipeNetId().set(netId);
```

Comment:
The placement of this seems kinda odd, although I'm on my phone which kinda sucks for reading comprehension